content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +35,643 @@ are covered here as well as the `System.Diagnostics` API.
+use a no-op implementation and fail to generate data.
 
-public static void ChildOperation()
-{
-    using var childActivity = MyActivitySource.StartActivity("ChildActivity");
+If you followed the instructions to [initialize the SDK](#initialize-the-sdk)
+above, you have a `TracerProvider` setup for you already. You can continue with
+[setting up an ActivitySource](#setting-up-an-activitysource).
 
-    // Track work in ChildOperation with childActivity
-}
-```
 
-When you view spans in a trace visualization tool, `ChildActivity` will be
-tracked as a nested operation under `ParentActivity`.
+### Setting up an ActivitySource
 
-### Nested Activities in the same scope
+Anywhere in your application where you write manual tracing code should configure an
+[`ActivitySource`](/docs/concepts/signals/traces/#tracer), which will be how you
+trace operations with [`Activity`](/docs/concepts/signals/traces/#spans)
+elements.
+
+It’s generally recommended to define `ActivitySource` once per app/service that is been instrumented, but you can instantiate several `ActivitySource`s if that suits your scenario.
 
-You may wish to create a parent-child relationship in the same scope. Although
-possible, this is generally not recommended because you need to be careful to
-end any nested `Activity` when you expect it to end.
+In the case of the example app, we will create a new file `Instrumentation.cs` as a custom type to hold reference for the ActivitySource. 
 
 ```csharp
-public static void DoWork()
-{
-    using var parentActivity = MyActivitySource.StartActivity("ParentActivity");
+using System.Diagnostics;
 
-    // Do some work tracked by parentActivity
+/// <summary>
+/// It is recommended to use a custom type to hold references for ActivitySource. 
+/// This avoids possible type collisions with other components in the DI container.
+/// </summary>
+public class Instrumentation : IDisposable
+{
+    internal const string ActivitySourceName = "dice-server";
+    internal const string ActivitySourceVersion = "1.0.0";
 
-    using (var childActivity = MyActivitySource.StartActivity("ChildActivity"))
+    public Instrumentation()
     {
-        // Do some "child" work in the same function
+        this.ActivitySource = new ActivitySource(ActivitySourceName, ActivitySourceVersion);
     }
 
-    // Finish up work tracked by parentActivity again
+    public ActivitySource ActivitySource { get; }
+
+    public void Dispose()
+    {
+        this.ActivitySource.Dispose();
+    }
 }
 ```
 
-In the preceding example, `childActivity` is ended because the scope of the
-`using` block is explicitly defined, rather than scoped to `DoWork` itself like
-`parentActivity`.
+Then we will update the `Program.cs` to add the Instrument object as a dependency injection:
+
+```csharp 
+//...
+
+// Register the ActivitySource as a singleton in the DI container.
+// Create a service to expose the ActivitySource Instruments.
```

Comment:
```suggestion
// Create a service to expose the ActivitySource.
```

Unless you plan to add metric instruments as well to Instrumentation.cs?

Reply:
Removed that line and updated the above comment.